include/CLI/impl/App_inl.hpp

```diff
@@ -1402,7 +1414,7 @@ CLI11_INLINE void App::_process_requirements() {
                 }
             }
         }
-        if(sub->count() > 0 || sub->name_.empty()) {
+        if(sub->count() > 0 || sub->name_.empty() && need_subcommands_.find(sub.get()) == need_subcommands_.end()) {
```

Comment:
this check is to process the requirements for the subcommands that were given but were not in the needed list